### PR TITLE
No logging when using default Value

### PR DESCRIPTION
**Pull request by lpriima**

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/Config.java [hunk 1]
```diff
@@ -1059,7 +1059,7 @@ private static String propertyNameToSystemPropertyName(final String setting) {
   private static <T> T valueOf(
       final String value, @NonNull final Class<T> tClass, final T defaultValue) {
     if (value == null || value.trim().isEmpty()) {
-      log.debug("valueOf: using defaultValue '{}' for '{}' of '{}' ", defaultValue, value, tClass);
+      log.trace("valueOf: using defaultValue '{}' for '{}' of '{}' ", defaultValue, value, tClass);
```
Comment (mar-kolya) [inline dd-trace-api/src/main/java/datadog/trace/api/Config.java:1062]: Are you sure this can be useful at all?

Comment (lpriima) [reply dd-trace-api/src/main/java/datadog/trace/api/Config.java:1062]: not sure. delete it ?

Comment (mar-kolya) [reply dd-trace-api/src/main/java/datadog/trace/api/Config.java:1062]: Personally I'd vote for not having it unless there is a specific problem it solves.

File: dd-java-agent/agent-profiling/profiling-controller/src/test/java/com/datadog/profiling/controller/ControllerFactoryTest.java [hunk 2]
```diff
@@ -28,9 +28,15 @@ public void testCreateControllerJava8() {
             });
     String expected =
         "The JFR controller could not find a supported JFR API, use OpenJDK 11+ or Azul zulu version 1.8.0_212+";
+    final String javaVendor = System.getProperty("java.vendor");
+    final String javaVersion = System.getProperty("java.version");
+    System.out.println(javaVendor + " " + javaVersion);
     if ("Azul Systems, Inc.".equals(System.getProperty("java.vendor"))) {
```
Comment (tylerbenson) [inline dd-java-agent/agent-profiling/profiling-controller/src/test/java/com/datadog/profiling/controller/ControllerFactoryTest.java:34]:
```suggestion
    if ("Azul Systems, Inc.".equals(javaVendor)) {
```

**Pull request merged**